actors/builtin/miner/miner_actor.go

```diff
@@ -1080,7 +1086,13 @@ func (a Actor) DeclareFaults(rt Runtime, params *DeclareFaultsParams) *adt.Empty
 	newFaultPowerTotal := NewPowerPairZero()
 	rt.State().Transaction(&st, func() interface{} {
 		info := getMinerInfo(rt, &st)
-		rt.ValidateImmediateCallerIs(info.Worker)
+		// if a key change is in progress, we also allow the owner to call this
+		// otherwise, we only allow the worker.
+		if info.PendingWorkerKey == nil {
```

Comment:
I assume we also need to allow declaring faults recovered?

Reply:
Done.

Reply:
That was not omitted from the description in #547 by accident. The ability to declare recoveries is not nearly as necessary as the ability to declare faults, and thus excluded. However, given other discussion there it might make sense to include it. We'll resolve it in the issue.